ibridges/irodsconnector/tickets.py

```diff
@@ -1,40 +1,43 @@
 """ ticket operations
 """
-import logging
-from random import choice
-from string import ascii_letters
-from subprocess import Popen, PIPE
-
 import irods.ticket
+from irods.models import TicketQuery
 
-from . import Icommands  # Seems to be not there anymore?
-from ibridges.irodsconnector import session
+from .session import Session
+import ibridges.irodsconnector.keywords as kw
+import warnings
+from typing import Optional
 
 
 class Tickets(object):
     """Irods Ticket operations """
 
-    def __init__(self, sess_man: session.Session):
+    def __init__(self, session: Session):
         """ iRODS data operations initialization
 
             Parameters
             ----------
-            sess_man : session.Session
+            session : session.Session
                 instance of the Session class
 
         """
-        self.sess_man = sess_man
+        self.session = session
+        self._all_tickets = None
```

Comment:
Shouldn't we initialize the tickets here? Or are you worried about performance?

Reply:
Good point!!!